jpa/src/main/java/org/jdbi/v3/jpa/JpaClass.java

```diff
@@ -0,0 +1,167 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.jpa;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.persistence.Column;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Transient;
+import java.beans.IndexedPropertyDescriptor;
+import java.beans.IntrospectionException;
+import java.beans.Introspector;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static java.util.Collections.unmodifiableList;
+
+class JpaClass<C> {
+
+    private static final Map<Class<?>, JpaClass<?>> cache = new ConcurrentHashMap<>();
+
+    public static <C> JpaClass<C> get(Class<C> clazz) {
+        return (JpaClass<C>) cache.computeIfAbsent(clazz, JpaClass::new);
+    }
+
+    private final List<JpaMember> members;
+
+    private JpaClass(Class<C> clazz) {
+        logger.debug("init {}", clazz);
+
+        this.members = unmodifiableList(new ArrayList<>(inspectClass(clazz)));
+
+        logger.debug("init {}: {} members.", clazz, members.size());
+    }
+
+    private static Collection<JpaMember> inspectClass(Class<?> clazz) {
+        Map<String, JpaMember> members = new HashMap<>();
+        List<String> transients = new ArrayList<>();
+
+        inspectFields(clazz, members, transients);
+        inspectAnnotatedProperties(clazz, members, transients);
+        inspectSuperclasses(clazz, members, transients);
+        inspectNonAnnotatedProperties(clazz, members, transients);
+        transients.forEach(members::remove);
+
+        return members.values();
+    }
+
+    private static void inspectSuperclasses(Class<?> clazz,
+                                            Map<String, JpaMember> members,
+                                            List<String> transients) {
+        while ((clazz = clazz.getSuperclass()) != null) {
+            if (clazz.isAnnotationPresent(MappedSuperclass.class)) {
+                inspectFields(clazz, members, transients);
+            }
+        }
+    }
+
+    private static void inspectFields(Class<?> clazz,
+                                      Map<String, JpaMember> members,
+                                      List<String> transients) {
+        for (Field member : clazz.getDeclaredFields()) {
+            if (member.getAnnotation(Transient.class) != null) {
+                transients.add(member.getName());
```

Comment:
Here I'm trying to match the admittedly confusing rules in JPA, in an effort to reduce pain for folks migrating away from JPA. See http://stackoverflow.com/questions/7016380/excluding-fields-from-embedded-properties-on-case-by-case-basis-with-hibernate

I think it makes sense to remove `@Transient` support for now. The rules about which annotation wins are not intuitive, and I'd rather not support it until we know there's a need for it.